operator/v1/types_clustermanager.go

```diff
@@ -132,8 +132,56 @@ type RegistrationDriverHub struct {
 	// AwsIrsa represents the configuration for awsirsa driver.
 	// +optional
 	AwsIrsa *AwsIrsaConfig `json:"awsirsa,omitempty"`
+
+	// GRPC represents the configuration for gRPC driver.
+	// +optional
+	GRPC *GRPCConfig `json:"grpc,omitempty"`
+}
+
+// GRPC represents the configuration for gRPC driver.
+type GRPCConfig struct {
+	// ImagePullSpec represents the desired image of the gRPC broker installed on hub.
+	// +optional
+	// +kubebuilder:default=quay.io/open-cluster-management/registration
+	ImagePullSpec string `json:"imagePullSpec,omitempty"`
+
+	// EndpointExposure represents the configuration for endpoint exposure.
+	// +optional
+	EndpointExposure *GRPCEndpointExposure `json:"endpointExposure,omitempty"`
+
+	// AutoApprovedIdentities represent a list of approved arn patterns
+	// +optional
+	AutoApprovedIdentities []string `json:"autoApprovedIdentities,omitempty"`
 }
 
+type GRPCEndpointExposure struct {
+	// Type specifies how the gRPC endpoint is exposed.
+	// You may need to apply an object to expose the gRPC endpoint, for example: a route.
+	// TODO: support loadbalancer.
+	// +kubebuilder:default:=hostname
+	// +kubebuilder:validation:Enum=hostname
+	// +required
+	Type GRPCEndpointExposureType `json:"type,omitempty"`
+
+	// Hostname points to a fixed hostname for serving agents' handshakes.
+	// +optional
```

Comment:
this may be required?

Reply:
It should be optional, considering we may introduce a LoadBalancer in the future.